qubesmanager/qube_manager.py

```diff
@@ -807,9 +810,88 @@ def __init__(self, qt_app, qubes_app, dispatcher, _parent=None):
 
         self.check_updates()
 
+    def change_template(self, template):
+        selected_vms = self.get_selected_vms()
+        reply = QMessageBox.question(
+            self, self.tr("Template Change Confirmation"),
+            self.tr("Do you want to change '{0}'<br>"
+                "to Template <b>'{1}'</b>?").format(
+                ', '.join(vm.name for vm in selected_vms), template),
+            QMessageBox.Yes | QMessageBox.Cancel)
+
+        if reply == QMessageBox.Yes:
+            failed = []
+            for info in selected_vms:
+                try:
+                    info.vm.template = template
+                except:
+                    failed.append(info.name)
+
+            if failed:
+                info_dialog = QMessageBox(self)
+                info_dialog.setWindowTitle(self.tr("Warning!"))
+                info_dialog.setText(
+                        self.tr("Some template change failed: {0} "
+                            ).format(", ".join(failed)))
+                info_dialog.show()
+
+    def change_network(self, netvm_name):
+        selected_vms = self.get_selected_vms()
+        reply = QMessageBox.question(
+            self, self.tr("Network Change Confirmation"),
+            self.tr("Do you want to change '{0}'<br>"
+                "to Network <b>'{1}'</b>?").format(
+                ', '.join(vm.name for vm in selected_vms), netvm_name),
+            QMessageBox.Yes | QMessageBox.Cancel)
+
+        if reply != QMessageBox.Yes:
+            return
+
+        try:
+            if netvm_name is not None:
+                check_power = any(info.state['power'] == 'Running' for info
+                        in self.get_selected_vms())
+                netvm = self.qubes_cache.get_vm(name=netvm_name)
+                if check_power and netvm.state['power'] != 'Running':
+                    reply = QMessageBox.question(
+                        self, self.tr("Qube Start Confirmation"),
+                        self.tr("<br>Can not change netvm to a halted Qube.<br>"
+                            "Do you want to start the Qube <b>'{0}'</b>?").format(
+                            netvm_name),
+                        QMessageBox.Yes | QMessageBox.Cancel)
+
+                    if reply == QMessageBox.Yes:
+                        with common_threads.busy_cursor():
+                            netvm.vm.start()
+                    else:
+                        return
+
+            failed = []
+            for info in self.get_selected_vms():
+                try:
+                    info.vm.netvm = netvm_name
+                except:
+                    failed.append(info.name)
```

Comment:
Can you also save error message?

Reply:
Yes, no problem.

Reply:
Now it does a new dialog per error. Maybe better if errors messages are too long, maybe worse if a lot of VM fail.

In any case I guess this will pretty rare to fail (if VM is started it does not let you to change the template) and when happens the error will be caught.